src/orion/algo/hyperband.py

```diff
@@ -349,22 +403,24 @@ def suggest(self, num):
                 f"{self.__class__.__name__} cannot suggest new samples and must wait "
                 "for trials to complete."
             )
-
+        # TODO: Shouldn't this return None in this case?
         return []
 
     @property
-    def executed_times(self):
+    def executed_times(self) -> int:
         """Counter for how many times Hyperband been executed"""
         if not self.brackets:
             return 0
         executed_times = self.brackets[-1].repetition_id
+        assert self.budgets is not None
         all_brackets_done = all(
             bracket.is_done for bracket in self.brackets[-len(self.budgets) :]
         )
         return executed_times - int(not all_brackets_done)
 
-    def _refresh_brackets(self):
+    def _refresh_brackets(self) -> None:
         """Refresh bracket if one hyperband execution is done"""
+        assert self.brackets is not None
```

Comment:
Another

Reply:
Fixed in 35ce6867769d77028c87f7bb015adfd349315f85